tsconfig.base.json

```diff
@@ -0,0 +1,33 @@
+{
+  "compilerOptions": {
```

Comment:
Do we need a typescript config? We aren't using a Typescript compiler, this will never be run.

Reply:
We need it because ESLint is using the TypeScript parser. You may be wondering why one would use the TypeScript parser. The reason is because we are leveraging several TypeScript lint rules to lint our JS. This does not imply that we intend to have any TypeScript sourcecode in this codebase, but rather that we are leveraging all of the code quality tooling of that ecosystem without necessarily having to author our modules in that language (but the option is still on the table).

Reply:
I took a look at the eslint repository, and this isn't the case. It is a special config for typescript repositories. Typescript is a potential extension. See https://github.com/eslint/eslint#what-about-experimental-features. I recommend taking a look at the repository, its pure js. 

Additionally, the config readme does not recommend this: https://eslint.org/docs/user-guide/configuring -- nowhere is the tsconfig mentioned.

I am really confused why we are over specifying things so much?

Reply:
Yes, I've seen the ESLint repo's codebase, it's very nice and worth revisiting. Thanks for reminding me of that.

The reason why things are overspecified and we can't just choose one of those very popular presets is because it would be extremely weird for TC39, the organization that specifies JS, to simply adopt Airbnb's or Google's code conventions since it would look like an endorsement for that company. At least that's how I see it. Maybe Jordan can clarify a bit better.

Reply:
ESLint init gives three choices:

![image](https://user-images.githubusercontent.com/17770407/96025735-8dc3dd80-0e23-11eb-8451-0bc7665c0a4c.png)

This config in this PR is actually using standard (actually semi-standard).

I'm going to have to see if I can trim down this config file down to not override any of that style guide's rules.

Reply:
Out of those three choices, you can look at github stars and npm downloads to see the relative popularity, which I'm pretty sure matches the order presented. I'm biased, but I'd suggest the first choice.

Reply:
"Standard" isn't what you want to use... The group that made it just named it "standard", it doesn't have anything to do with standards or industry agreement. Please see their readme: https://github.com/standard/standard#but-this-isnt-a-real-web-standard -- additionally, the practices that they encourage, such as leaving out semicolons, means that parsers do double work when parsing ambiguous statements.

The eslint config here is not quite right. Its a mix between the website and your crawler. We don't use esm on the website, or typescript. For the crawler it also doesn't fit either, as your crawler doesn't run in the browser. 

you can have multiple eslint configs within the same project. A base config for the website, which should have no options beyond the default. Then for your crawler you can make a separate config, which will be defaulted to by the directory structure. These are the options you want to use:

```
Website config:
✔ How would you like to use ESLint? · problems
✔ What type of modules does your project use? · none
✔ Which framework does your project use? · none
✔ Does your project use TypeScript? · **No** / Yes
✔ Where does your code run? · browser
✔ What format do you want your config file to be in? · JSON
Local ESLint installation not found.
The config that you've selected requires the following dependencies:
```

The resulting config file, which is the only file we need in this pr, is this:

.eslintrc.json 
```
{
    "env": {
        "browser": true,
        "es2021": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 12
    },
    "rules": {
    }
}
```

That's it. It should be in the top level directory of this project. The only other changes should be npm modules that are added. Nothing else is necessary. This is all I expect from this pull request. 

Ok, for your crawler, you can create an eslint config file specific to it, and put it in the folder with your crawler. ESLint is cascading, and this config will override the base config. You can read more here: https://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy . Then, when you are doing work on your crawler code, eslint will fall back on the nearest eslint config, which will be the one you specify. For this, please do it separate from this pull request -- and make it part of the pull request for your crawler.

Reply:
There is a slight discrepancy here that I would like to note with the config you proposed above. For the browser code, you are configuring it to run as though it is ES2021, but for compatability reasons that cannot be the case. This is going to need to use [`babel-preset-env`](https://babeljs.io/docs/en/babel-preset-env) + [Browserslist](https://babeljs.io/docs/en/babel-preset-env#browserslist-integration). Shouldn't be a problem, though.